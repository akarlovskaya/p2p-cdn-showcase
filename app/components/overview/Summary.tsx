import React from "react";

type SummaryProps = {
  userRole: string;
  total: number;
  currentMonth: string;
};

function Summary({ userRole, total, currentMonth }: SummaryProps) {
  return (
    <div className="text-center">
      <h2 className="text-2xl text-black font-bold mt-8 mb-2">
        {userRole === "client" ? "Balance" : "Earnings"} in {currentMonth}
      </h2>
      <p className="font-mono text-6xl text-black font-bold mb-4">${total}</p>
    </div>
  );
}

export default Summary;
