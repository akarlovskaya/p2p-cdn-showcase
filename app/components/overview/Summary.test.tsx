import React from "react";
import { render, screen } from "@testing-library/react";
import Summary from "./Summary";

describe("Summary Component", () => {
  // Test 1: Renders correctly for client role
  it("displays total balance for client role", () => {
    render(<Summary userRole="client" total={1000} currentMonth="April" />);

    expect(screen.getByText("Total Balance in April")).toBeInTheDocument();
    expect(screen.getByText("$1000")).toBeInTheDocument();
  });

  // Test 2: Renders correctly for provider role
  it("displays total earnings for provider role", () => {
    render(<Summary userRole="provider" total={2000} currentMonth="May" />);

    expect(screen.getByText("Total Earnings in May")).toBeInTheDocument();
    expect(screen.getByText("$2000")).toBeInTheDocument();
  });

  // Test 3: Formats total number correctly
  it("formats the total amount correctly", () => {
    render(<Summary userRole="client" total={1234567.89} currentMonth="May" />);

    expect(screen.getByText("$1234567.89")).toBeInTheDocument();
  });

  // Test 4: Handles zero total
  it("displays zero total correctly", () => {
    render(<Summary userRole="provider" total={0} currentMonth="May" />);

    expect(screen.getByText("$0")).toBeInTheDocument();
  });
});

// Snapshot test
it("matches snapshot", () => {
  const { asFragment } = render(
    <Summary userRole="client" total={1000} currentMonth="May" />
  );
  expect(asFragment()).toMatchSnapshot();
});
