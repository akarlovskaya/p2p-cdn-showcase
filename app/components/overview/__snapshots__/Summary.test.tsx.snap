// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="text-center"
  >
    <h2
      class="text-2xl text-black font-bold mt-8 mb-2"
    >
      Total Balance in May
    </h2>
    <p
      class="font-mono text-6xl text-black font-bold mb-4"
    >
      $1000
    </p>
  </div>
</DocumentFragment>
`;
